File: services/api-aspnet-core/App/Compliance.cs
using System.Text.Json;

namespace App;

// Industry compliance profile, read at startup.
// One repo serves every industry; COMPLIANCE_PROFILE picks one profile
// out of compliance/profiles.json. The controls flip at boot, no rebuild.
public class Compliance
{
    public string Profile { get; }
    public string Name { get; }
    public string Jurisdiction { get; }

    // Raw control values for the active profile. Values are bool/number/string,
    // kept as JsonElement so /compliance serves them with their original JSON type.
    public IReadOnlyDictionary<string, JsonElement> Controls { get; }

    // JWT lifetime in seconds. Driven by the profile's session_timeout_seconds
    // control (HIPAA/ITSG-33 → 900s). A 0 or missing value means "no profile
    // limit", so fall back to the prior 8-hour default.
    public long SessionTimeoutSeconds
    {
        get
        {
            if (Controls.TryGetValue("session_timeout_seconds", out var v)
                && v.ValueKind == JsonValueKind.Number
                && v.TryGetInt64(out var seconds)
                && seconds > 0)
            {
                return seconds;
            }
            return 8 * 60 * 60;
        }
    }

    public Compliance()
    {
        var profile = (Environment.GetEnvironmentVariable("COMPLIANCE_PROFILE") ?? "baseline").Trim();

        var path = ResolveProfilesPath();
        if (path is null)
        {
            Console.Error.WriteLine("FATAL: compliance/profiles.json not found");
            Environment.Exit(1);
        }

        Catalog catalog;
        try
        {
            using var stream = File.OpenRead(path!);
            catalog = JsonSerializer.Deserialize<Catalog>(stream, JsonOptions)
                ?? throw new InvalidOperationException("profiles.json deserialized to null");
        }
        catch (Exception e)
        {
            Console.Error.WriteLine($"FATAL: compliance/profiles.json not loadable: {e.Message}");
            Environment.Exit(1);
            throw; // unreachable; keeps the compiler happy about non-null fields below.
        }

        if (!catalog.Profiles.TryGetValue(profile, out var active))
        {
            Console.Error.WriteLine($"FATAL: unknown COMPLIANCE_PROFILE: {profile}");
            Environment.Exit(1);
            throw new InvalidOperationException();
        }

        Profile = profile;
        Name = active.Name;
        Jurisdiction = active.Jurisdiction;
        Controls = active.Controls;
    }

    // Look for compliance/profiles.json next to the running app, then in the
    // working directory, then by walking up the tree (tests run from bin/).
    private static string? ResolveProfilesPath()
    {
        var roots = new[] { AppContext.BaseDirectory, Directory.GetCurrentDirectory() };
        foreach (var root in roots)
        {
            var dir = new DirectoryInfo(root);
            while (dir is not null)
            {
                var candidate = Path.Combine(dir.FullName, "compliance", "profiles.json");
                if (File.Exists(candidate)) return candidate;
                dir = dir.Parent;
            }
        }
        return null;
    }

    private static readonly JsonSerializerOptions JsonOptions = new()
    {
        PropertyNameCaseInsensitive = true,
    };

    private sealed class Catalog
    {
        public Dictionary<string, ProfileEntry> Profiles { get; set; } = new();
    }

    private sealed class ProfileEntry
    {
        public string Name { get; set; } = "";
        public string Jurisdiction { get; set; } = "";
        public Dictionary<string, JsonElement> Controls { get; set; } = new();
    }
}

File: services/api-rails/lib/compliance.rb
# Industry compliance profile, read at startup.
# One repo serves every industry; COMPLIANCE_PROFILE picks one of the
# generated profiles in compliance/profiles.json. The controls flip at
# boot, no rebuild.
require 'json'

module Compliance
  CATALOG_FILE = 'profiles.json'.freeze

  def self.active
    @active ||= load_profile
  end

  # Test hook — drop the memoized profile so a new ENV value is picked up.
  def self.reset!
    @active = nil
  end

  def self.catalog_path
    base = defined?(Rails) && Rails.respond_to?(:root) && Rails.root ? Rails.root.to_s : Dir.pwd
    candidate = File.join(base, 'compliance', CATALOG_FILE)
    return candidate if File.exist?(candidate)

    # Fall back to the working directory when not running under Rails.
    File.join(Dir.pwd, 'compliance', CATALOG_FILE)
  end

  def self.load_catalog
    path = catalog_path
    JSON.parse(File.read(path))
  rescue StandardError => e
    warn "FATAL: compliance catalog not loadable: #{path}: #{e.message}"
    exit 1
  end

  def self.load_profile
    name = ENV['COMPLIANCE_PROFILE'] || 'baseline'
    catalog = load_catalog
    profiles = catalog['profiles'] || {}

    entry = profiles[name]
    unless entry
      warn "FATAL: unknown COMPLIANCE_PROFILE: #{name}"
      exit 1
    end

    {
      profile: name,
      name: entry['name'],
      jurisdiction: entry['jurisdiction'],
      controls: entry['controls'] || {}
    }
  end
end
